src/ml/neural_net/style_transfer/mps_style_transfer_backend.mm

```diff
@@ -0,0 +1,145 @@
+/* Copyright © 2019 Apple Inc. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-3-clause license that can
+ * be found in the LICENSE.txt file or at https://opensource.org/licenses/BSD-3-Clause
+ */
+
+#import <ml/neural_net/style_transfer/mps_style_transfer_backend.hpp>
+#import <ml/neural_net/style_transfer/mps_style_transfer.h>
+#import <ml/neural_net/mps_utils.h>
+
+@interface TCMPSStyleTransferHelpers:NSObject
++(float_array_map) fromNSDictionary: (NSDictionary<NSString *, NSData *> *) dictionary;
++(NSDictionary<NSString *, NSData *> *) toNSDictionary: (float_array_map) map;
+@end
+
+@implementation TCMPSStyleTransferHelpers
++(float_array_map) fromNSDictionary: (NSDictionary<NSString *, NSData *> *) dictionary {
+  float_array_map map;
+  for (NSString* key in dictionary) {
+    NSData* value = [dictionary objectForKey:key];
+
+    size_t dataLength = (size_t) (value.length / sizeof(float));
+    std::vector<size_t> dataShape = { dataLength }; 
+    float *dataBytes = (float *) (value.bytes);
+    
+    shared_float_array array = shared_float_array::copy(dataBytes, dataShape);
+
+    map.insert(std::pair<std::string, shared_float_array>([key UTF8String], array));  
+  }
+  return map;
+}
+
++(NSDictionary<NSString *, NSData *> *) toNSDictionary: (float_array_map) map {
+  NSMutableDictionary<NSString *, NSData *> * dictionary 
+      = [[NSMutableDictionary alloc] init];
+
+  for (const auto &element : map) {
+    const float* elementData = element.second.data();
+    const size_t elementSize = element.second.size() * sizeof(float);
+
+    NSMutableData * mutableElementData = [NSMutableData dataWithLength:elementSize];
+    memcpy(mutableElementData.mutableBytes, elementData, elementSize);
```

Comment:
For now, these two lines should just be
```
NSData *data = [NSData dataWithBytes:elementData length:elementSize];
```

Let's also add a TODO here. The issue you file for wrapping `NSData` with a `float_array` subclass can go the other way, too. We can construct an `NSData` that wraps a `shared_float_array`, using this API: https://developer.apple.com/documentation/foundation/nsdata/1417337-initwithbytesnocopy?language=objc

Let's not do this now, since it's not on the critical path to testing the MPS implementation. But we should avoid these copies in general before we release.